backend/internal/service/handler/spotistics/clientcreds.go

```diff
@@ -0,0 +1,51 @@
+package spotistics
+
+import (
+	"context"
+	"log"
+	"os"
+
+	"github.com/gofiber/fiber/v2"
+	spotifyauth "github.com/zmb3/spotify/v2/auth"
+
+	"github.com/zmb3/spotify/v2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+type SpotifyCredentials struct {
+	ClientID     string
+	ClientSecret string
+}
+
+type SpotifyKey struct{}
+
+func (creds *SpotifyCredentials) getClient() spotify.Client {
```

Comment:
```suggestion
func (creds *SpotifyCredentials) getClient(ctx context.Context) (spotify.Client , error) {
```
this is called from the middleware where we have a context, let's drill it down into this func